Dockerfile

```diff
@@ -0,0 +1,30 @@
+FROM ruby:2.7.8-slim-bullseye as base
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    build-essential \
+    git \
+    curl \
+    && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /site
+
+COPY Gemfile /site/Gemfile
+COPY Gemfile.lock /site/Gemfile.lock
+
+RUN gem update --system && bundle install && gem cleanup
```

Comment:
I'm not super familiar with the base ruby image, but in my environment, the system gems are provided by system RPMs in my distribution. This has been my experience using CentOS/RHEL/Fedora. So, a `yum` (or `dnf`) update is normally sufficient and I don't need to do a `gem update`. Is the situation different for the base ruby image?

Also, the instructions on the README in https://github.com/apache/accumulo-website#local-builds-for-testing explain in detail how to set up the GEM environment as a non-supervisor user for local gems installed to the user directory by bundler. I think it would be helpful if this Dockerfile did something similar, just automated, so instead of setting things up differently by doing system installed gems. That way, there's only a single set of instructions to follow, whether we're doing something via a local build, or trying to maintain this Dockerfile in future. However, it looks like this is relying on a different set of environmental assumptions from the those in the README. It looks like this is installing gems using the superuser, and gems are being installed to a system directory rather than a local user directory. I'm not 100% certain of this, since I'm not super familiar with Docker. If this is diverging from the detailed README instructions, it would be helpful to have some inline comments explaining what's being done here, so we have a better chance of being able to maintain it going forward.

Reply:
> I'm not super familiar with the base ruby image, but in my environment, the system gems are provided by system RPMs in my distribution. This has been my experience using CentOS/RHEL/Fedora. So, a `yum` (or `dnf`) update is normally sufficient and I don't need to do a `gem update`. Is the situation different for the base ruby image?
> 
The ruby image uses RubyGems for dependency management. In the Dockerfile we run the `apt-get update` command first to ensure that system packages are updated and also install some basic required packages. 

`gem update --system` ensures that any Gems loaded from RubyGems are also updated before we attempt to install our application specific gems via `bundle install`. 

Your comment got me curious so I popped into the image and verified there are no ruby-related packages installed via `apt`.  

> Also, the instructions on the README in https://github.com/apache/accumulo-website#local-builds-for-testing explain in detail how to set up the GEM environment as a non-supervisor user for local gems installed to the user directory by bundler. I think it would be helpful if this Dockerfile did something similar, just automated, so instead of setting things up differently by doing system installed gems. That way, there's only a single set of instructions to follow, whether we're doing something via a local build, or trying to maintain this Dockerfile in future. However, it looks like this is relying on a different set of environmental assumptions from the those in the README. It looks like this is installing gems using the superuser, and gems are being installed to a system directory rather than a local user directory. I'm not 100% certain of this, since I'm not super familiar with Docker. If this is diverging from the detailed README instructions, it would be helpful to have some inline comments explaining what's being done here, so we have a better chance of being able to maintain it going forward.

Docker is operating on it's own separate filesystem layer so nothing installed in the container will affect anything on the given system. This accomplishes the same separation as setting `GEM_HOME` to a user directory. 

If you're interested, the [Ruby image spec](https://hub.docker.com/_/ruby) calls out various environment variables that are set in the image, including `GEM_HOME` which is set to `/usr/local/bundle`. 

The explicit `gem install bundler` command isn't needed as the Gemfile.lock requires Ruby 2.7 to be the 
minimum installed and bundler was added as a default gem in 2.6. 

So, I don't see this diverging from the detailed readme instructions as `GEM_HOME` is already set in the container and the next command to run is `bundle install`. 

As long as all of the required dependencies are defined in the Gemfile, then we can ensure that all development environments are held to the same standard. 

The **only** action that would require an update to the Dockerfile is a Ruby version update. But that is still extremely simple as you would just update the `FROM` image tag and run the documented `_/scripts/build-images.sh` script .

Reply:
> Docker is operating on it's own separate filesystem layer so nothing installed in the container will affect anything on the given system. This accomplishes the same separation as setting `GEM_HOME` to a user directory.

My main concern isn't separation between the container filesystem and the local filesystem, but the user being able to comprehend the analogous behavior inside the container to the local system. Specifically, in the local system, there is a separation between the OS's package manager and the gem package management by setting `GEM_HOME` so that the gem package manager stores stuff in a different location than the locations where RPM or DEB packages get installed. This separation is comprehensible in the README because `GEM_HOME` is explicitly set. It's not clear how this separation occurs inside the container.

> 
> If you're interested, the [Ruby image spec](https://hub.docker.com/_/ruby) calls out various environment variables that are set in the image, including `GEM_HOME` which is set to `/usr/local/bundle`.

Perfect. In that case, I think all that's needed to help connect the behavior of this container to the instructions in the README is to add a simple comment above the gem commands that mention something like: `GEM_HOME is set by the container`. You probably don't even need to specify that it's `/usr/local/bundle`, because that could change. Just knowing that it's already set should be enough to connect one's understanding of the README to what's going on in the container.

> 
> The explicit `gem install bundler` command isn't needed as the Gemfile.lock requires Ruby 2.7 to be the minimum installed and bundler was added as a default gem in 2.6.

Not really relevant for this PR, but just to highlight this challenge for some: Annoyingly, in Fedora 37, even though ruby.rpm is 3.1 and rubygems.rpm is 3.3, rubygems-bundler.rpm still ships in a separate optional RPM, even though it's supposed to be a default gem since 2.6 and should be included with rubygems.rpm. I think there was a discussion about fixing this, but I lost track of the discussion.

> 
> So, I don't see this diverging from the detailed readme instructions as `GEM_HOME` is already set in the container and the next command to run is `bundle install`.

Now that I understand what it's doing (thanks for digging in and explaining it), I agree it's not different. I think just a simple comment can help connect the dots for future devs trying to figure this out.

Reply:
Sounds good! I'll put that comment in tomorrow and get this closed up.

Reply:
Comment has been added